### ISSUE TITLE
Add disk to powered on kvm fails.

### ISSUE THREAD

**Issue by gurevichmark** [xcat2/xcat-core#1628]
Currently adding a disk to a kvm (`chvm -a`) works for a kvm that is **not** powered on. But fails for a kvm that is already powered on.

**Comment by immarvin** [comment]
hi @gurevichmark , maybe you can try  ``virsh attach-disk <domain> <source> <target>``. the <source> is the full path of the qcow2 file on the kvm host. the <target> is the disk device name for the newly added disk on the kvm guest, you should make sure the specified <target> does not exist on the kvm guest.

**Comment by gurevichmark** [comment]
@immarvin Thank you for the suggestion, I tried `virsh attach-disk <domain> <source> <target>` and the results are the same as with `virsh attach-device <xml file>`. 
```
[root@fs2 ~]# virsh attach-disk mgXgeneratedvm1 /var/lib/libvirt/images/mgXgeneratedvm1.sdb.qcow2 sdb
error: Failed to attach disk
error: operation failed: target sdb already exists

[root@fs2 ~]# virsh attach-disk mgXgeneratedvm1 /var/lib/libvirt/images/mgXgeneratedvm1.sdd.qcow2 sdd
error: Failed to attach disk
error: Cannot access storage file '/var/lib/libvirt/images/mgXgeneratedvm1.sdd.qcow2' (as uid:107, gid:107): No such file or directory

[root@fs2 ~]#
```

I also tried running it with a flag `--live`, same result.

**Comment by immarvin** [comment]
hi @gurevichmark , would you please run``virsh dumpxml mgXgeneratedvm1 `` on the kvm host or "fdisk -l" inside the kvm guest, is there already a disk named "sdb" in the kvm guest?

For "/var/lib/libvirt/images/mgXgeneratedvm1.sdd.qcow2", how was this qcow2 file produced?

**Comment by gurevichmark** [comment]
I think the problem might be with the `target` we assign to the disk to be attached. It seems that when a vm is created target `sda` is for a disk, and `sdb` is for cdrom. However when `chvm -a <size>` is issued on this vm, we attempt to create a disk with `sdb` target, even though `sdb` is already assigned to the cdrom. 
I think the solution should be to fix `chvm -a` processing to skip the target assigned to cdrom.

**Comment by gurevichmark** [comment]
The root cause seems to me is `get_multiple_paths_by_url()` in kvm.pm.
When `chvm -a <size>` is issued, the `chvm()` tries to figure out what letter to use for suffix of the new disk - `b`, `c`, `d`, etc. It does that by calling `get_multiple_paths_by_url()` which returns a hash of existing volumes on the VM with its target name - `sdb`,`sdc`,`sdd`, etc. The existing volumes are retrieved by using `$poolobj->list_volumes()`, that however, does not include `cdrom`. 
As a result `sdb` is not included in the list returned from `get_multiple_paths_by_url()`, so `chvm` assigns the new disk to `sdb` even though `cdrom` is already using that target.

**Comment by gurevichmark** [comment]
Similar problem exists if we first do `chvm -a <size>` on the powered **off** VM. The disk gets added to the `kvm_nodedata` table with `sdb` diskname. Then, when we poweron the VM, error is reported that there is a conflict of 2 devices with `sdb` name. The root cause is probably the same as above, name of the cdrom is not considered.

**Comment by gurevichmark** [comment]
Fixed by #1751
